Rejig layout so that main does not contain nav

danacotoran opened the pull request:
## What
Restructured the markup so that the main content of the page on the account manager does not contain navigation.
Achieved this by moving the grid layout into the application layout file. The account navigation was moved into a `account_navigation` content block which, if present, is inserted before `main` in the application layout file. 
The `main` element is now a child element of the grid instead of containing the grid as it did before.


## Why
The account manager has a "Skip to main content" link, whose role is to help users skip over the navigation and reach the primary content on the page. 
The purpose of this navigation mechanism is to help the user jump over the navigation content and straight to the main information. However, in our case, the main information itself contains navigation content.


### Before: navigation inside `main`
<img width="1354" alt="Screenshot 2021-01-04 at 14 54 55" src="https://user-images.githubusercontent.com/7116819/104163496-6cf13100-53ee-11eb-8518-a1d69d0263dd.png">

### After: navigation outside `main`
<img width="1499" alt="Screenshot 2021-01-11 at 09 23 35" src="https://user-images.githubusercontent.com/7116819/104163698-bf325200-53ee-11eb-88db-55fea63623b7.png">


https://trello.com/c/hw4HVWsC/542-main-content-should-not-contain-navigation

andysellick commented on app/views/account/show.html.erb:
```diff
@@ -1,75 +1,72 @@
 <% content_for :title, t("account.your_account.heading") %>
 <% content_for :location, "your-account" %>
+<% content_for :account_navigation do %>
+<div class="govuk-grid-column-one-third">
```
Nitpick: indent lines 4-6.

andysellick commented on app/views/account/show.html.erb:
```diff
@@ -1,75 +1,72 @@
 <% content_for :title, t("account.your_account.heading") %>
 <% content_for :location, "your-account" %>
+<% content_for :account_navigation do %>
+<div class="govuk-grid-column-one-third">
```
All of the `account_navigation` blocks seem to contain the same `govuk-grid-column-one-third`. Could this be moved to the parent so as to not be repeated throughout lots of templates?

danacotoran replied:
Good point! Have done this 👍

andysellick commented on app/views/edit_phone/resend.html.erb:
```diff
@@ -1,21 +1,17 @@
-      margin_top: 0,
-      margin_bottom: 3,
-    } %>
+<%= render "govuk_publishing_components/components/heading", {
```
Maybe not for this PR, but since every page like this has the same format of heading, I wonder if a further optimisation could be made by declaring the component use at the top level template (or a partial) and passing only the required text to that.

danacotoran replied:
Good point, that would reduce repetition even further. There are some differences in heading format between some of the files though (mostly to do with margins it seems?) but maybe we can do some work to unify them anyway.

andysellick commented on app/views/edit_phone/resend.html.erb:
```diff
@@ -1,21 +1,17 @@
-  </div>
-</div>
+<%= form_with(url: edit_user_registration_phone_code_path, method: :post) do %>
+<%= render "govuk_publishing_components/components/button", {
```
Nitpick: indent lines 14-16.

andysellick commented on app/views/manage/show.html.erb:
```diff
@@ -1,107 +1,104 @@
 <% content_for :title, t("account.manage.heading") %>
 <% content_for :location, "manage" %>
+<% content_for :account_navigation do %>
+<div class="govuk-grid-column-one-third">
```
Nitpick: indent links 4-6.

andysellick commented on app/views/devise/registrations/edit_password.html.erb:
```diff
@@ -1,71 +1,68 @@
-    <%= email_field_tag 'email', resource.email, class: "govuk-!-display-none", autocomplete: "off", "aria-hidden": true %>
-    <%= render "devise/shared/error_messages", resource: resource %>
+<%= form_for(resource, as: resource_name, url: registration_path(resource_name), html: { method: :put }) do %>
+<%= email_field_tag 'email', resource.email, class: "govuk-!-display-none", autocomplete: "off", "aria-hidden": true %>
```
Nitpick: indent lines 18-19.

andysellick reviewed (commented):
This is a good change, although I think the description for the first commit could be more explicit about the change being made, for example explaining the implementation of the `account_navigation` block and moving the rest of the grid layout into the parent templates for most of this code.

danacotoran commented:
@andysellick thanks for the comments, I've updated the PR based on your suggestions. I've added a bit more detail to the first commit, hopefully it is clearer now.

andysellick reviewed (approved):
Good stuff.
